src/frontend/src/pages/GanttPage/GanttChartPage.tsx

```diff
@@ -199,13 +203,66 @@ const GanttChartPage: FC = () => {
   const teamList = Array.from(new Set(projects.map(getProjectTeamsName)));
   const sortedTeamList: string[] = teamList.sort(sortTeamNames);
 
-  // do something here with the data
-  const saveChanges = (eventChanges: EventChange[]) => {
-    if (eventChanges.length === 0) {
-      console.log('no changes do nothing');
-    } else {
-      console.log('Changes:', eventChanges);
+  const updateEventChange = (eventId: string, newRequestEventChange: RequestEventChange) => {
+    setGroupedEventChanges((prevMap) => {
+      const updatedMap = new Map(prevMap);
+      updatedMap.set(eventId, newRequestEventChange);
+      return updatedMap;
+    });
+  };
+
+  const updateEventDates = (event: any, change: EventChange, existingChange?: any) => {
+    let newStart = existingChange?.newStart || event.start;
+    let newEnd = existingChange?.newEnd || event.end;
+    let duration = existingChange?.duration || daysBetween(event.end, event.start);
+
+    switch (change.type) {
+      case 'change-end-date':
+        duration = daysBetween(change.newEnd, change.originalEnd);
+        newEnd = addDaysToDate(newStart, duration);
+        break;
+      case 'shift-by-days':
+        newStart = addDaysToDate(event.start, change.days);
+        newEnd = addDaysToDate(newStart, duration);
+        break;
     }
+
+    return { newStart, newEnd, duration };
+  };
+
+  const saveChanges = (eventChanges: EventChange[]) => {
+    eventChanges.forEach((change) => {
+      const event = ganttTasks.find((task) => task.id === change.eventId);
+      if (event) {
+        const existingChange = groupedEventChanges.get(change.eventId);
+        console.log('Existing Change', existingChange);
+
+        const { newStart, newEnd, duration } = updateEventDates(event, change, existingChange);
+
+        const newEventChange = {
+          eventId: change.eventId,
+          name: event.name,
+          prevStart: event.start,
+          prevEnd: event.end,
+          newStart,
+          newEnd,
+          duration
+        };
+        updateEventChange(change.eventId, newEventChange);
+      }
+    });
+
+    // Log and update active modals
+    console.log('Grouped event changes: ', groupedEventChanges);
+    const updatedSet = new Set(activeModalIds);
+    groupedEventChanges.forEach((change) => updatedSet.add(change.eventId));
+    setActiveModalIds(updatedSet);
+  };
+
+  const removeActiveModal = (changeId: string) => {
+    const updatedSet = new Set(activeModalIds);
+    updatedSet.delete(changeId);
+    setActiveModalIds(updatedSet);
```

Comment:
Ignore if you've already done it but ensure updatedSet is actually updating like u expect it to